### PR TITLE
[Mac] Use static registrar!

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/build/MacOSX/monostub.m [hunk 1]
```diff
@@ -1,404 +0,0 @@
-//gcc -m32 monostub.m -o monostub -framework AppKit
```
Comment (Therzok) [inline main/build/MacOSX/monostub.m]: this is a symlink so we don't break existing code.

**Comment by slluis** [comment]
What's this change for?

**Comment by Therzok** [comment]
View the PR by-commit, as the github UI can't understand what went on with the symlink and rename detection fails.

**Comment by Therzok** [comment]
The first commit now introduces an explanation of what this change does

**Comment by Therzok** [comment]
Rebased. The previous build only had debugger unit tests failures which are unrelated. Merging.

**Pull request merged**